### ISSUE TITLE
[4.0.0-beta.2] Elementsources in project.yaml are not respected when applying the project config

### ISSUE THREAD

**Issue by wsydney76** [craftcms/cms#10774]
### Description

> Element source settings are now stored in the project config.

Exited about that, as it saves us from writing migrations.

However those settings are not (yet) respected when applying the project config against a fresh install: 
Also tried `craft project-config/apply --force`

This is in project.yaml

````
elementSources:
  craft\elements\Asset:
    -
      key: 'volume:5a0be839-14a2-465e-a504-bb63dd0281e9' # Images
      tableAttributes:
        - imageSize
        - dateModified
        - link
      type: native
    -
      heading: Internal
      type: heading
    -
      key: 'volume:3b60e4f9-36bb-4456-bee5-67160b5a3108' # User Photos
      tableAttributes:
        - filename
        - size
        - dateModified
        - uploader
        - link
      type: native
    -
      key: 'volume:ed627edd-73bd-4b75-8684-f410bf4e4b9d' # Temp Images
      type: native
  craft\elements\Entry:
    -
      key: '*'
      tableAttributes:
        - section
        - postDate
        - author
        - link
      type: native
    -
      key: singles
      type: native
    -
      key: 'section:84b98802-2548-4aed-b67e-fc97f6ed9e30' # Page
      tableAttributes:
        - drafts
        - type
        - 'field:7df417e8-1eea-46d8-85a7-2a20370a881e' # Featured Image
        - postDate
        - author
        - link
      type: native
````

The element index still shows the default:

![image](https://user-images.githubusercontent.com/60010597/158798476-312f7340-c6d0-43ed-843b-e5b0ec51c978.png)

### Steps to reproduce

1. see above

### Additional info

- Craft version: 4.0.0-beta.2
- PHP version: 8.0.15
- Database driver & version:
- Plugins & versions: none

**Comment by brandonkelly** [comment]
Hm… are you sure the project config changes have been applied? Craft 4 _only_ stores element source settings in the project config, so they don’t need to be synced to the database like most things.

**Comment by wsydney76** [comment]
@brandonkelly This is just a fresh install: 

`composer install` with ` "craftcms/cms": "4.0.x-dev"`, followed by `php craft install` with an existing project config containing the element sources above.

There is no `elementSources` key  in the Project Config utility  'Loaded Project Config Data' though.

**Comment by wsydney76** [comment]
@brandonkelly Thanks for looking into it. However, I can't get it to work on my local Windows machine, looks like there might be some OS or version dependency thing.

> Was your elementSources data present in project.yaml before you ran the installer?
Yes

> At the very end, do you see this as the second-to-last thing output?
Yes

So just created a new test case:

Fresh 3.7 install, added a field, a section, und changed element index settings, upgraded to `4.0.0-beta.3`

Looks fine and pushed that to https://github.com/wsydney76/craft-issue-10774/tree/main

If I install this on nitro using `nitro create https://github.com/wsydney76/craft-issue-10774 temp1` everything is ok.

![image](https://user-images.githubusercontent.com/60010597/159681376-c0881a63-bf5e-4a17-add4-67be84c52d24.png)


On Windows/PHP 8.0.15 however:

Deleted all tables und ran `php craft install` again.

The element sources are gone and deleted from `project.yaml`

![Screenshot 2022-03-23 103630](https://user-images.githubusercontent.com/60010597/159672461-69b9577c-80d8-4339-8bb2-20398a3e3959.jpg)

Pushed that to a reinstall branch: https://github.com/wsydney76/craft-issue-10774/blob/reinstall/config/project/project.yaml

**Comment by brandonkelly** [comment]
And Craft 4 Beta was in place when you ran `php craft install`?

**Comment by wsydney76** [comment]
> And Craft 4 Beta was in place when you ran php craft install?

Yes, see composer.lock


````
 {
            "name": "craftcms/cms",
            "version": "4.0.0-beta.3",
            "source": {
                "type": "git",
                "url": "https://github.com/craftcms/cms.git",
                "reference": "e8bf3d348fac4cf1a2999ff09b1317ff4f1cd5f7"
            },
````
After install:

![image](https://user-images.githubusercontent.com/60010597/159723662-e72d943c-fddf-4924-bd2e-7bb546324451.png)

**Comment by wsydney76** [comment]
It's ok to leave this unresolved for now, as it may be just a local thing on my machine.

Just finished my task of evaluating and preparing Craft 4 migration and handed the results over to my former agency. They will follow up if needed.

So that's the end of my temporary comeback to the IT world, now back to retirement in a sunny spring in a world gone mad... 

Thanks a lot for all your help.

**Comment by brandonkelly** [comment]
Ahh! Thanks for investigating! Just fixed this for the next release.

**Comment by brandonkelly** [comment]
Craft 4.0.4 is out now with that fix. (Fix was also done for 3.7.44 just in case something else ends up adding slashes to project config paths there.)
